app/helpers/sessions_helper.rb

```diff
@@ -38,7 +39,30 @@ def clear_cart
     end
     
     def log_out
+        user = User.find_by(id: session[:user_id])
+        user.update_attribute(:cart, hto_s(session[:cart]))
```

Comment:
Let's get a test for this at some point in `test/helpers/sessions_helper`. 
Also, best to familiarize yourselves with json methods, since we'll probably be doing a lot of that later.
```ruby
require 'json'
test "should make string from cart hash" do
  cart_hash = { :x1 => 2, :x2 => 1 }
  assert_equal "x1 2,x2 1", hto_s(cart_hash)
  assert_equal '{"x1":2,"x2":1}', { :x1 => 2, :x2 => 1 }.to_json   # always true
end

test "should make hash from string" do
  hash = JSON.parse("someJSONstring")
end
```